src/Datadog.Trace/AppSec/Waf/NativeBindings/Native.cs

```diff
@@ -39,43 +57,200 @@ internal class Native
+                        ?? new List<string>();
+
+            // the real trace home
+            var tracerHome = Environment.GetEnvironmentVariable("DD_DOTNET_TRACER_HOME");
+            if (!string.IsNullOrWhiteSpace(tracerHome))
+            {
+                integrationsPaths.Add(tracerHome);
+            }
+
+            // include the appdomain base as this will help framework samples running in IIS find the library
+            var currentDir =
+                string.IsNullOrEmpty(AppDomain.CurrentDomain.RelativeSearchPath) ?
+                    AppDomain.CurrentDomain.BaseDirectory : AppDomain.CurrentDomain.RelativeSearchPath;
+            integrationsPaths.Add(currentDir);
+
+            // the home folder could contain the native dll directly, but it could also
+            // be under a runtime specific folder
+            var paths =
+                integrationsPaths
+                    .Distinct()
+                    .SelectMany(GetRuntimeSpecificVersions)
+                    .ToList();
+
+            return paths;
+        }
+
+        private static void SearchPathsAndLoadLibrary(string libName, List<string> paths)
+        {
+            var success = false;
+            foreach (var path in paths)
+            {
+                var libFullPath = Path.Combine(path, libName);
+
+                if (!File.Exists(libFullPath))
+                {
+                    continue;
+                }
+
+                // loading the library is sufficient, once in memory the p/invokes will just work
+                var loaded = NativeLibrary.TryLoad(libFullPath, out var _);
+
+                if (loaded)
+                {
+                    success = true;
+                    Log.Information($"Loaded library '{libName}' from '{path}'");
+                    break;
+                }
+                else
+                {
+                    Log.Warning($"Failed to load library '{libName}' from '{path}'");
+                }
+            }
+
+            if (!success)
+            {
+                Log.Warning($"Failed to load library '{libName}' from any of the following '{string.Join(", ", paths)}'");
+            }
+        }
+
+        private static void GetLibNameAndRuntimeId(out string libName, out string runtimeId)
+        {
+            string runtimeIdPart1, libPrefix, libExt;
+
+            switch (Environment.OSVersion.Platform)
+            {
+                case PlatformID.MacOSX:
+                    runtimeIdPart1 = "osx";
+                    libPrefix = "lib";
+                    libExt = "dylib";
+                    break;
+                case PlatformID.Unix:
+                    runtimeIdPart1 = "linux";
+                    libPrefix = "lib";
+                    libExt = "so";
+                    break;
+                case PlatformID.Win32NT:
+                case PlatformID.Win32S:
+                case PlatformID.Win32Windows:
+                case PlatformID.WinCE:
```

Comment:
We only support `Win32NT` out of these four.

Reply:
Changed.